vulkan/vulkan_proc_table.cc

```diff
@@ -145,42 +146,23 @@ bool VulkanProcTable::SetupDeviceProcAddresses(
   ACQUIRE_PROC(SetBufferCollectionConstraintsFUCHSIAX, handle);
   ACQUIRE_PROC(GetBufferCollectionPropertiesFUCHSIAX, handle);
 #endif  // OS_FUCHSIA
+#endif  // TEST_VULKAN_PROCS
   device_ = {handle, nullptr};
   return true;
 }
 
-bool VulkanProcTable::OpenLibraryHandle() {
+bool VulkanProcTable::OpenLibraryHandle(const char* path) {
 #if VULKAN_LINK_STATICALLY
-  static char kDummyLibraryHandle = '\0';
-  handle_ = reinterpret_cast<decltype(handle_)>(&kDummyLibraryHandle);
-  return true;
+  handle_ = fml::NativeLibrary::CreateForCurrentProcess();
 #else   // VULKAN_LINK_STATICALLY
-  dlerror();  // clear existing errors on thread.
-  handle_ = dlopen("libvulkan.so", RTLD_NOW | RTLD_LOCAL);
-  if (handle_ == nullptr) {
-    FML_DLOG(WARNING) << "Could not open the vulkan library: " << dlerror();
-    return false;
-  }
-  return true;
+  handle_ = fml::NativeLibrary::Create(path);
 #endif  // VULKAN_LINK_STATICALLY
+  return !!handle_;
```

Comment:
lol, I never thought of this to avoid an explicit bool cast without actually using an explicit cast. Love it.